collection_mutation.cc

```diff
@@ -350,7 +350,7 @@ difference(collection_mutation_view_description a, collection_mutation_view_desc
             ++it;
         }
         if (it == b.cells.end() || !key_type.equal(it->first, c.first)
-            || compare_atomic_cell_for_merge(c.second, it->second) > 0) {
+            || compare_atomic_cell_for_merge(c.second, it->second) >= 0) {
```

Comment:
This change will make this emit a difference for cells that compare equal (`0` -> equal).
The correct change is:
`compare_atomic_cell_for_merge(c.second, it->second) != std::strong_orderging::equal)`.

The current code is wrong because it only emits diff for cells which compare less, but not for those that compare greater.

Reply:
It must include the case where it is equal to 0. If it is changed to `compare_atomic_cell_for_merge(c.second, it->second) != std::strong_ordering::equal`, the test will fail.
```bash
        for row in cql.execute(f"SELECT * FROM ks.mv_cf_view WHERE base_key = 5 AND view_key = 5;"):
            print(row)
            assert row.base_key == 5
            assert row.view_key == 5
&gt;           assert row.attrs['attr1'] == 'aaaaa'
E           TypeError: 'NoneType' object is not subscriptable

```
This approach you provided does not solve the problem of missing attrs.

Reply:
I'm still not sure about this change, I want to spend some time to understand the connection of this comparison and the attribute loss by running the test locally.
I will be on holiday for the next two weeks, so this will only happen next year.

In the meanwhile I would appreciate a second review from @nyh or @piodul and team.

Reply:
Happy holiday for @denesb . Another important point is that the non-primary key columns of the created table must include columns of collection type. I saw in https://github.com/scylladb/scylladb/blob/master/db/view/view.cc#L1054 that there will be some optimizations for the collection columns in the future, and now it seems like it is something that must be done.
```bash
        //TODO(sarna): Optimize collections case - currently they do not go under optimization
        if (!cdef.is_atomic()) {
            return false;
        }
```